code/modules/awaymissions/mission_code/ghost_role_spawners/golems.dm

```diff
@@ -134,6 +134,24 @@
 		user.death()
 		return
 
+/obj/effect/mob_spawn/human/golem/attackby(obj/item/slimepotion/transference/I, mob/user, params)
```

Comment:
You need to make sure the object that is attacking is actually a transference potion, otherwise you can just hit the shell with any item to transfer yourself.

Reply:
I thought that placing it that way in the attackby parameters made it so that it only checked for the transference potion?

Reply:
Parameters themselves have no type checking. You could pass.. say an `/area` into this proc and `I` would be that area. Writing out the full type like you've done, does allow you to do `istype(I)`, instead of 
`istype(I, /obj/item/slimepotion/transference)` though.